Mage/src/main/java/mage/target/targetpointer/NthTargetPointer.java

```diff
@@ -0,0 +1,161 @@
+ */
+public abstract class NthTargetPointer extends TargetPointerImpl {
+
+    private static final Map<UUID, Integer> emptyZoneChangeCounter = Collections.unmodifiableMap(new HashMap<>(0));
+    private static final List<UUID> emptyTargets = Collections.unmodifiableList(new ArrayList<>(0));
+
+    private Map<UUID, Integer> zoneChangeCounter;
+    private int targetNumber;
+
+    public NthTargetPointer(int targetNumber) {
+        super();
+        this.targetNumber = targetNumber;
+    }
+
+    public NthTargetPointer(final  NthTargetPointer nthTargetPointer) {
+        super(nthTargetPointer);
+        this.targetNumber = nthTargetPointer.targetNumber;
+
+        if (nthTargetPointer.zoneChangeCounter != null) {
+            this.zoneChangeCounter = new HashMap<>(nthTargetPointer.zoneChangeCounter.size());
+            for (Map.Entry<UUID, Integer> entry : nthTargetPointer.zoneChangeCounter.entrySet()) {
+                addToZoneChangeCounter(entry.getKey(), entry.getValue());
+            }
+        }
+    }
+
+    @Override
+    public void init(Game game, Ability source) {
+        if (source.getTargets().size() < targetNumber) {
+            return;
+        }
+
+        for (UUID target : source.getTargets().get(targetIndex()).getTargets()) {
+            Card card = game.getCard(target);
+            if (card != null) {
+                addToZoneChangeCounter(target, card.getZoneChangeCounter(game));
+            }
+        }
+    }
+
+    @Override
+    public List<UUID> getTargets(Game game, Ability source) {
+        if (source.getTargets().size() < targetNumber) {
+            return emptyTargets;
+        }
+
+        List<UUID> targetIds = source.getTargets().get(targetIndex()).getTargets();
+        List<UUID> finalTargetIds = new ArrayList<>(targetIds.size());
+
+        for (UUID targetId : targetIds) {
+            Card card = game.getCard(targetId);
+            if (card != null
+                    && getZoneChangeCounter().containsKey(targetId)
+            && card.getZoneChangeCounter(game) != getZoneChangeCounter().get(targetId)) {
+                Permanent permanent = game.getPermanentOrLKIBattlefield(targetId);
+                if (permanent == null || permanent.getZoneChangeCounter(game) != getZoneChangeCounter().get(targetId)) {
+                    continue;
+                }
+            }
+
+            finalTargetIds.add(targetId);
+        }
+        return finalTargetIds;
+    }
+
+    @Override
+    public UUID getFirst(Game game, Ability source) {
+        if (source.getTargets().size() < targetNumber) {
+            return null;
+        }
+
+        UUID targetId = source.getTargets().get(targetIndex()).getFirstTarget();
+        if (getZoneChangeCounter().containsKey(targetId)) {
+            Card card = game.getCard(targetId);
+            if (card != null && getZoneChangeCounter().containsKey(targetId)
+                    && card.getZoneChangeCounter(game) != getZoneChangeCounter().get(targetId)) {
```

Comment:
You've retained the additional code using `game.getPermanentOrLKIBattlefield` that was in `FirstTargetPointer.getTargets` but you've removed the equivalent code from this method. Was that intentional? Also, I don't think removing the comment explaining why that strange-looking code exists is a good idea.

Reply:
Was not intentional, oversight on my part. I've added that and added the comments back in.